### PR TITLE
Support pragma once in all header files and cleanup some warnings

### PR DESCRIPTION
**Pull request by cngzhnp**
As you know, almost all compilers support "pragma once" keyword instead of using include guards. To be keep consistency between header files, all header files are edited.

Besides this, try to fix some warnings about loss of data.

### CODE REVIEW

File: db/version_set.cc [hunk 1]
```diff
@@ -895,13 +895,16 @@ void Version::GetColumnFamilyMetaData(ColumnFamilyMetaData* cf_meta) {
...
+          MakeTableFileName("", file->fd.GetNumber()),
+          file_path,
+          static_cast<size_t>(file->fd.GetFileSize()),
+          file->fd.smallest_seqno,
+          file->fd.largest_seqno,
+          file->smallest.user_key().ToString(),
           file->largest.user_key().ToString(),
           file->stats.num_reads_sampled.load(std::memory_order_relaxed),
-          file->being_compacted);
+          file->being_compacted });
```
Comment (ajkr) [inline db/version_set.cc]: oops extra space

File: port/dirent.h [hunk 2]
```diff
@@ -42,6 +41,4 @@ using port::closedir;
 
 }  // namespace rocksdb
 
-#endif  // OS_WIN
-
-#endif  // STORAGE_LEVELDB_PORT_DIRENT_H_
+#endif  // OS_WIN
```
Comment (ajkr) [inline port/dirent.h]: minor: the newline before EOF is missing

File: third-party/fbson/FbsonUtil.h [hunk 3]
```diff
@@ -158,6 +157,4 @@ class FbsonToJson {
   char buffer_[OUT_BUF_SIZE];
 };
 
-} // namespace fbson
-
-#endif // FBSON_FBSONUTIL_H
+} // namespace fbson
```
Comment (ajkr) [inline third-party/fbson/FbsonUtil.h]: minor: newline before EOf

File: util/crc32c_ppc.h [hunk 4]
```diff
@@ -18,6 +17,4 @@ extern uint32_t crc32c_ppc(uint32_t crc, unsigned char const *buffer,
 
 #ifdef __cplusplus
 }
-#endif
-
-#endif
+#endif
```
Comment (ajkr) [inline util/crc32c_ppc.h]: again

File: util/file_reader_writer.cc [hunk 5]
```diff
@@ -694,11 +694,11 @@ Status FilePrefetchBuffer::Prefetch(RandomAccessFileReader* reader,
     buffer_.Alignment(alignment);
     buffer_.AllocateNewBuffer(static_cast<size_t>(roundup_len),
                               copy_data_to_new_buffer, chunk_offset_in_buffer,
-                              chunk_len);
+			                        static_cast<size_t>(chunk_len));
```
Comment (ajkr) [inline util/file_reader_writer.cc]: minor: convert tabs to spaces

File: util/transaction_test_util.cc [hunk 6]
```diff
@@ -258,7 +259,7 @@ Status RandomTransactionInserter::Verify(DB* db, uint16_t num_sets,
   std::vector<uint16_t> set_vec(num_sets);
   std::iota(set_vec.begin(), set_vec.end(), static_cast<uint16_t>(0));
   if (rand) {
-    std::random_shuffle(set_vec.begin(), set_vec.end(),
+		std::random_shuffle(set_vec.begin(), set_vec.end(),
```
Comment (ajkr) [inline util/transaction_test_util.cc]: tabs again

File: utilities/blob_db/blob_db_impl.cc [hunk 7]
```diff
@@ -1041,7 +1041,7 @@ Status BlobDBImpl::GetBlobValue(const Slice& key, const Slice& index_entry,
   }
   Slice crc_slice(blob_record.data(), sizeof(uint32_t));
   Slice blob_value(blob_record.data() + sizeof(uint32_t) + key.size(),
-                   blob_index.size());
+		               static_cast<size_t>(blob_index.size()));
```
Comment (ajkr) [inline utilities/blob_db/blob_db_impl.cc]: tabs

**Comment by cngzhnp** [comment]
I tried to fix the minor issues, hope that everythıng seems OK now.

**Pull request closed without merging**